clouds/tests/aws/test_aws_utils.py

```diff
@@ -0,0 +1,123 @@
+from typing import Dict, List
+from unittest.mock import patch
+
+from clouds.aws.aws_utils import (
+    set_and_verify_aws_credentials,
+    set_and_verify_aws_config,
+    delete_all_objects_from_s3_folder,
+    delete_bucket,
+    aws_region_names,
+    get_least_crowded_aws_vpc_region,
+)
+
+os_environ_patch_str: str = "clouds.aws.aws_utils.os.environ"
+config_parser_patch_str: str = "clouds.aws.aws_utils.ConfigParser"
+ec2_client_patch_str: str = "clouds.aws.aws_utils.ec2_client"
+botocore_client_patch_str: str = "clouds.aws.aws_utils.botocore.client"
+
+dummy_bucket_name: str = "test_bucket"
+us_east_region_str: str = "us-east-1"
+us_west_region_str: str = "us-west-1"
+default_aws_config_section: str = "default"
+
+
+@patch(os_environ_patch_str)
+@patch(config_parser_patch_str)
+@patch(ec2_client_patch_str)
+def test_set_and_verify_aws_credentials(mock_ec2_client, mock_config_parser, mock_os_environ, mocker):
+    mock_os_environ.get.return_value = None
+
+    mock_parser_instance = mocker.MagicMock()
+    mock_config_parser.return_value = mock_parser_instance
+    mock_parser_instance.get.side_effect = ["dummy_access_key", "dummy_secret_key"]
+
+    mock_ec2_client_instance = mocker.MagicMock()
+    mock_ec2_client.return_value = mock_ec2_client_instance
+    mock_ec2_client_instance.describe_regions.return_value = {"Regions": [{"RegionName": us_east_region_str}]}
```

Comment:
should be in fixtures

Reply:
maybe we should move only mock/mocked objects to fixtures and keep side_effect and return values setup in the tests since it is specific to the test itself , wyt?

Reply:
@rnetser please approve